### ISSUE TITLE
AP: interpret unlisted as public

### ISSUE THREAD

**Issue by snarfed** [snarfed/bridgy-fed#1036]
We currently don't, but it sounds like we should. Background:
* https://docs.joinmastodon.org/spec/activitypub/#Mention
* https://zirk.us/@philippsteinkrueger/112411487158100467
* https://seb.jambor.dev/posts/understanding-activitypub/#unlisted-messages

**Comment by qazmlp** [comment]
`@phillippsteinkrueger@zirk.us` isn't quite right about this, as there are a few other important effects of 'Unlisted':

- removes the post from remote 'Federated timeline's (and some other app-specific discovery features).
- removes the post from 'Search' (including hashtag search).

Essentially, it turns off "Bluesky-style" interaction modes, if you will.
I use Unlisted posting specifically to avoid these, and many other people do too.

I would not be able to use the bridge at all if this was changed globally.
(Making it a per-account setting somehow would be a-okay though, as long as it's opt-in.)

---

*That said*, there are some good arguments for bridging *some* Unlisted posts over to Bluesky in my eyes, as long as either it's a thread and the first post in the thread was Public (this is a decently common style of "less-annoying" threading on the fediverse, as Public replies do appear in most aggregate feeds there) or the post @-mentions a bridged Bluesky user, so that the bridging intention can be assumed for opted-in authors.

Bridging Unlisted replies from opted-in users where all previous posts in the thread are visible to Bluesky would probably be fine too, since Bluesky's default behaviour for replies actually resembles Mastodon's *somewhat*. It still exposes them to search iinm, but that should be a smaller issue than how incomplete discussions would be without that.

**Comment by snarfed** [comment]
Interesting! Thanks for the details.

**Comment by ygg2** [comment]
In addition to a replies setting, maybe a setting for bridging Unlisted boosts as well?
I heard Akkoma has boosts as Unlisted or at least has that as a default

**Comment by snarfed** [comment]
Hmm, interesting! 

Honestly, user-level settings for anything are pretty unlikely. They add a level of complexity, UX, auth, and maintenance and compatibility burden that I'm not quite ready to take on. I definitely appreciate the ideas and info though!

**Comment by ygg2** [comment]
> Honestly, user-level settings for anything are pretty unlikely. They add a level of complexity, UX, auth, and maintenance and compatibility burden that I'm not quite ready to take on. I definitely appreciate the ideas and info though!

Makes sense. Perhaps there could be a case like "bridge Unlisted boosts of posts that are already bridged/are originally Bluesky posts"?

I think the main use of Unlisted boosting is to avoid spamming timelines and the uses @qazmlp pointed out for Unlisted posting don't apply (booster isn't the initial poster) but I may be forgetting something

**Comment by qazmlp** [comment]
I agree, as long as the original post has been bridged already (or it's a self-boost of a Public post), it makes complete sense to me to bridge Unlisted boosts.

**Comment by snarfed** [comment]
@ygg2 interesting, good catch! Thanks for reporting. Looking at the AS2 for https://sakurajima.social/notes/9w68fc7934 , it is indeed unlisted: it's `to: https://sakurajima.social/users/9r3pm4h264/followers` and only `cc: https://www.w3.org/ns/activitystreams#Public`. Hrm.

**Comment by ygg2** [comment]
Small addition to this, it seems like Unlisted self-boost doesn't work the same. I had to un-boost and then re-boost as Public.

**Comment by DenebTM** [comment]
Until (if ever) this behaviour is changed, could the fact that Unlisted posts aren't bridged be mentioned in https://fed.brid.gy/docs#fediverse-what ?

**Comment by snarfed** [comment]
@DenebTM Yes! Good idea, will do.

**Comment by Tamschi** [comment]
I think this is causing confusion again since Mastodon renamed 'unlisted'.
It would be good to list 'quiet public' explicitly under [#visibility](https://fed.brid.gy/docs#visibility).

**Comment by snarfed** [comment]
Thanks! Agreed, done, deploying now.

**Comment by imsodin** [comment]
Have you considered the previously suggested option ([latter part of this comment](https://github.com/snarfed/bridgy-fed/issues/1036#issuecomment-2102970963)) to conditionally bridge unlisted posts, if they are in response to a public/bridged post? Or potentially simpler heuristic still, any unlisted post that @-mentions a bluesky account (resp. whatever is the protocol equivalent of that)?  
Concretely what I am curious about is: Do you think that's a sensible thing to do in principle? And if so, can you think of technical challenges off the top of your head to implementing it?  
I am aware that even if your answers were Yes and No, that wouldn't mean anything about it happening (prios, limited time, personal interest, ... - any of all the usual reasons, valid by definition because personal). I am mostly wanting to gauge the feasibility and potential acceptance of such a change in case "anyone" wanted to tackle it (don't expect anything please, just very, very maybe :) ).

I just started trying out the bridge (1) and stumbled upon this when my unlisted responses didn't turn up in bluesky. I usually respond unlisted, unless I feel like I have something to say worth reading by anyone outside of the thread, which is almost never. In practice it doesn't matter much, I am mostly just a lurker, and I could post public instead as a workaround, but I still would like to follow that principles just in case. Plus I don't see any reason why a user wouldn't want a response to a bluesky post (aka they opted in already) to be visible alongside that bluesky post.

(1) Recently there's interesting content on bluesky I don't find in the fediverse, so I started using it again. However I like the client experience with phanpy a lot more and would prefer just a single social feed/app, thus considering the bridge again. Also having a lot of hope around the opt-out discussions going on in other tickets at the moment, as that's a currently a significant barrier to a unified experience for me.

**Comment by snarfed** [comment]
Honestly, I'm not sure.

I definitely get how bridging unlisted/quiet public posts would be useful! And even other non-public features like DMs too, #1425. However, when I first started building https://brid.gy/ , the predecessor to Bridgy Fed, over 13y ago (!), I drew a bright line that it only handles fully public content, and I continued that with Bridgy Fed. [As just one person's side project, with no funding or organizational structure or other protection,](https://snarfed.org/2024-11-01_53932) that's made me a bit more comfortable that I'm minimizing one big source of risk, harm, and liability.

So, while I get that unlisted is still technically public, and would be very useful to bridge, I'm still reluctant.

**Comment by snarfed** [comment]
Apart from that, the drawbacks in https://github.com/snarfed/bridgy-fed/issues/1036#issuecomment-2102970963 still seem to apply. We don't have a way to recreate the [fediverse's unlisted/quiet public behavior](https://docs.joinmastodon.org/user/posting/#unlisted), ie hiding from feeds and search, in Bluesky. If we bridge unlisted posts, even replies, they'll still immediately show up in Bluesky custom feeds and search. That mismatch isn't great.

**Comment by makew0rld** [comment]
On my instance, the convention is to use unlisted for replies to the first post in your thread. This prevents long threads from clogging up the local feed. Since Bluesky has no local feed, it makes sense to bridge unlisted posts (IMO) bc otherwise my threads are broken.

Tusky (popular fedi Android client) even supports doing this automatically for replies to yourself, so I don't think it's so uncommon.

**Comment by ygg2** [comment]
Now that there are per-user config options, is bridging Quiet Public replies on the table for a future preference, or not really due to the bridge flag or something else?

**Comment by snarfed** [comment]
It's definitely on the table. We're still reluctant to start proliferating per-user settings, even with the new web login and UI on https://fed.brid.gy/ , so a setting probably wouldn't be our first approach here, but we're open to proposals and (then) PRs.

**Comment by snarfed** [comment]
One place to start here could be allowlisting this for specific instances where it makes sense, eg https://bird.makeup/

**Comment by TomCasavant** [comment]
If you don't end up bridging unlisted posts, could you add some sort of notification from the bridgy account to alert someone when they send an unlisted reply to bridged account? (I've accidentally replied to bluesky posts without changing it to public several times w/o realizing)

**Comment by ianh** [comment]
I am running into this issue trying to bridge a bot from fediverse to bsky.  The instance rules require that bot posts be marked as limited, so as it is, it's impossible to bridge the account while following the instance rules.
